sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.CosmosDB.Models;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+
+namespace Azure.ResourceManager.CosmosDB.Tests.ScenarioTests
+{
+    [TestFixture]
+    public class GraphResourcesOperationsTests : CosmosDBManagementClientBase
+    {
+        protected string resourceGroupName;
+        protected string databaseAccountName;
+        protected string databaseName = "databaseName1002";
+        protected string databaseName2 = "databaseName21002";
+        protected string gremlinGraphName = "gremlinGraphName1002";
+
+        protected string graphThroughputType = "Microsoft.DocumentDB/databaseAccounts/gremlinDatabases/throughputSettings";
+        protected int sampleThroughput = 700;
+
+        public GraphResourcesOperationsTests()
+            : base(true)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                InitializeClients();
+                resourceGroupName = Recording.GenerateAssetName(CosmosDBTestUtilities.ResourceGroupPrefix);
+                databaseAccountName = Recording.GenerateAssetName("amegraphtest");
+                await CosmosDBTestUtilities.TryRegisterResourceGroupAsync(ResourceGroupsOperations, CosmosDBTestUtilities.Location, resourceGroupName);
+            }
+        }
+
+        [TearDown]
+        public async Task CleanupResourceGroup()
+        {
+            await CleanupResourceGroupsAsync();
+        }
+
+        [TestCase]
+        public async Task GraphCRUDTests()
+        {
+            var locations = new List<Location>();
+            Location loc = new Location();
+            loc.LocationName = CosmosDBTestUtilities.Location;
+            loc.FailoverPriority = 0;
+            loc.IsZoneRedundant = false;
+            locations.Add(loc);
+
+            DatabaseAccountCreateUpdateParameters databaseAccountCreateUpdateParameters = new DatabaseAccountCreateUpdateParameters(locations);
+            databaseAccountCreateUpdateParameters.Location = CosmosDBTestUtilities.Location;
+            databaseAccountCreateUpdateParameters.Capabilities.Add(new Capability("EnableGremlin"));
+            databaseAccountCreateUpdateParameters.ConsistencyPolicy = new ConsistencyPolicy(DefaultConsistencyLevel.Eventual);
+
+            await WaitForCompletionAsync(await CosmosDBManagementClient.DatabaseAccounts.StartCreateOrUpdateAsync(resourceGroupName, databaseAccountName, databaseAccountCreateUpdateParameters));
+
+            Task<Response> taskIsDatabaseNameExists = CosmosDBManagementClient.DatabaseAccounts.CheckNameExistsAsync(databaseAccountName);
```

Comment:
Just use await here, then you won't need to do all these extra steps you are doing

Reply:
resolved